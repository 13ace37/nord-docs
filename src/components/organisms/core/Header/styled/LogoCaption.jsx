/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import styled from "styled-components";

import { motion, ms } from "styles/theme";

/**
 * Caption for Nord's logo component.
 * It will fade out when the header changes into pinned (collapsed) mode.
 *
 * @author Arctic Ice Studio <development@arcticicestudio.com>
 * @author Sven Greb <development@svengreb.de>
 * @since 0.3.0
 */
const LogoCaption = styled.span`
  font-size: ${ms(6)};
  font-weight: 500;
  transition: opacity ${motion.speed.duration.transition.text.fade}ms ease-in-out;
  opacity: ${({ isPinned }) => isPinned && 0};
  margin-left: 0.2775em;
`;

export default LogoCaption;
