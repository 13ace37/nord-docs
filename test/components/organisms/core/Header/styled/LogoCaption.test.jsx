/*
 * Copyright (C) 2018-present Arctic Ice Studio <development@arcticicestudio.com>
 * Copyright (C) 2018-present Sven Greb <development@svengreb.de>
 *
 * Project:    Nord Docs
 * Repository: https://github.com/arcticicestudio/nord-docs
 * License:    MIT
 */

import React from "react";

import { renderWithTheme } from "nord-docs-test-utils";
import LogoCaption from "organisms/core/Header/styled/LogoCaption";

describe("behavior", () => {
  test("is invisible in pinned Mode", () => {
    const { container } = renderWithTheme(<LogoCaption isPinned />);
    expect(container.firstChild).not.toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("is visible in unpinned Mode", () => {
    const { container } = renderWithTheme(<LogoCaption isPinned={false} />);
    expect(container.firstChild).toBeVisible();
    expect(container.firstChild).toMatchSnapshot();
  });
});
