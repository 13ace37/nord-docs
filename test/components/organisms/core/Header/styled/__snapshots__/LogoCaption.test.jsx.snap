// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`behavior is invisible in pinned Mode 1`] = `
.c0 {
  font-size: 2.0272865295410156em;
  font-weight: 500;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  opacity: 0;
  margin-left: 0.2775em;
}

<span
  class="c0"
/>
`;

exports[`behavior is visible in unpinned Mode 1`] = `
.c0 {
  font-size: 2.0272865295410156em;
  font-weight: 500;
  -webkit-transition: opacity 200ms ease-in-out;
  transition: opacity 200ms ease-in-out;
  margin-left: 0.2775em;
}

<span
  class="c0"
/>
`;
